### PR TITLE
airflow2 migration

### LINKED ISSUE #22: Migrate to Airflow 2.x
The new major Airflow version [is released](http://airflow.apache.org/blog/airflow-two-point-oh-is-here/).
Need to migrate plugin to it.

### PR DESCRIPTION
**Pull request by owlleg6**
Helm chart version 8.5.2
close https://github.com/odahu/odahu-airflow-plugin/issues/22
Depends on PR https://github.com/odahu/odahu-airflow-plugin/pull/42

### CODE REVIEW

File: terraform/modules/k8s/airflow/main/templates/airflow.yaml [hunk 1]
```diff
@@ -11,19 +11,30 @@ flower:
...
+      from flask_appbuilder.security.manager import AUTH_DB
+
+      # the SQLAlchemy connection string
+      SQLALCHEMY_DATABASE_URI = conf.get('core', 'SQL_ALCHEMY_CONN')
+
+      # use embedded DB for auth
+      AUTH_TYPE = AUTH_DB
+
+      # setup Public role name, no authentication needed
+      AUTH_ROLE_PUBLIC = 'Admin'
```
Comment (karbyshevds) [inline terraform/modules/k8s/airflow/main/templates/airflow.yaml:30]: You can't remove auth and allow anonymous access, please fix this.

File: terraform/modules/k8s/airflow/main/templates/airflow.yaml [hunk 2]
```diff
@@ -11,19 +11,30 @@ flower:
...
+  webserverConfig:
+    stringOverride: |-
+      from airflow import configuration as conf
+      from flask_appbuilder.security.manager import AUTH_DB
+
+      # the SQLAlchemy connection string
+      SQLALCHEMY_DATABASE_URI = conf.get('core', 'SQL_ALCHEMY_CONN')
+
+      # use embedded DB for auth
+      AUTH_TYPE = AUTH_DB
```
Comment (karbyshevds) [inline terraform/modules/k8s/airflow/main/templates/airflow.yaml:27]: You shouldn't use auth_db, you should use SSO

File: terraform/modules/k8s/airflow/main/templates/airflow.yaml [hunk 3]
```diff
@@ -42,35 +53,40 @@ airflow:
           name: airflow-postgres
           key: postgresql-password
 
+  usersUpdate:
+    enabled: false
```
Comment (karbyshevds) [inline terraform/modules/k8s/airflow/main/templates/airflow.yaml:58]: Why did you disable users update?

**Review by karbyshevds — CHANGES_REQUESTED** [review]

File: terraform/modules/k8s/airflow/main/templates/airflow.yaml (hunk 3 above)
Comment (owlleg6) [reply terraform/modules/k8s/airflow/main/templates/airflow.yaml:58]: fixed

Comment (owlleg6) [reply terraform/modules/k8s/airflow/main/templates/airflow.yaml:58]: As final decision, we disabled usersUpdate and set airflow.users to an empty array

**Review by karbyshevds — APPROVED** [review]

File: terraform/modules/k8s/airflow/main/templates/airflow.yaml (hunk 2 above)
Comment (owlleg6) [reply terraform/modules/k8s/airflow/main/templates/airflow.yaml:27]: We decided to create separate task for implementation of remote_user

File: terraform/modules/k8s/airflow/main/templates/airflow.yaml (hunk 1 above)
Comment (owlleg6) [reply terraform/modules/k8s/airflow/main/templates/airflow.yaml:30]: We decided to create separate task for auth

**Review by GennadiyShpak — APPROVED** [review]

**Review by BPylypenko — APPROVED** [review]

**Review by lev-titov — APPROVED** [review]

**Pull request merged**